### PR TITLE
feat(rust): `list_indices` in remote SDK

### PR DESCRIPTION
**Pull request by wjones127**
Implements `list_indices`.

### CODE REVIEW

File: rust/lancedb/src/remote/table.rs [hunk 1]
```diff
@@ -525,11 +525,57 @@ impl<S: HttpSend> TableInternal for RemoteTable<S> {
...
+        let body = response.text().await?;
+        let body: ListIndicesResponse = serde_json::from_str(&body).map_err(|err| Error::Http {
+            message: format!(
+                "Failed to parse index list response: {}, body: {}",
+                err, body
+            ),
+        })?;
+
+        // Make request to get stats for each index, so we get the index type.
+        // This is a bit inefficient, but it's the only way to get the index type.
```
Comment (wjones127) [inline rust/lancedb/src/remote/table.rs:557]: TODO here: https://github.com/lancedb/sophon/issues/2607

File: rust/lancedb/src/remote/table.rs [hunk 2]
```diff
@@ -525,11 +525,57 @@ impl<S: HttpSend> TableInternal for RemoteTable<S> {
...
+        // Make request to list the indices
+        let request = self
+            .client
+            .post(&format!("/v1/table/{}/index/list/", self.name));
+        let response = self.client.send(request, true).await?;
+        let response = self.check_table_response(response).await?;
+
+        #[derive(Deserialize)]
+        struct ListIndicesResponse {
+            indices: Vec<IndexConfigResponse>,
```
Comment (wjones127) [inline rust/lancedb/src/remote/table.rs:539]: The API actually spells this as "indexes":
```suggestion
            indexes: Vec<IndexConfigResponse>,
```

File: rust/lancedb/src/remote/table.rs [hunk 3]
```diff
@@ -525,11 +525,57 @@ impl<S: HttpSend> TableInternal for RemoteTable<S> {
...
+        #[derive(Deserialize)]
+        struct IndexConfigResponse {
+            index_name: String,
+            columns: Vec<String>,
+        }
+
+        let body = response.text().await?;
+        let body: ListIndicesResponse = serde_json::from_str(&body).map_err(|err| Error::Http {
+            message: format!(
+                "Failed to parse index list response: {}, body: {}",
```
Comment (westonpace) [inline rust/lancedb/src/remote/table.rs:551]:
```suggestion
                "Failed to parse list_indices response: {}, body: {}",
```

File: rust/lancedb/src/remote/table.rs [hunk 4]
```diff
@@ -525,11 +525,57 @@ impl<S: HttpSend> TableInternal for RemoteTable<S> {
...
+        #[derive(Deserialize)]
+        struct ListIndicesResponse {
+            indexes: Vec<IndexConfigResponse>,
+        }
+
+        #[derive(Deserialize)]
+        struct IndexConfigResponse {
+            index_name: String,
+            columns: Vec<String>,
+        }
```
Comment (westonpace) [inline rust/lancedb/src/remote/table.rs:546]: Did not know you could do local structs like this, cool.

**Review by westonpace — APPROVED** [review]

**Pull request merged**